File: typescript/modules/libs/workspace/domain/system/src/lib/iworkspace.ts
import { IRule } from './derivation/rules/irule';
import { IConfiguration } from './iconfiguration';
import { ISession } from './isession';

export interface IWorkspace {
  configuration: IConfiguration;

  rules: IRule[];

  rule(cls: new (...args: any[]) => IRule): IRule;

  createSession(): ISession;
}

File: dotnet/System/Workspace/Allors.Workspace.Domain/IWorkspace.cs
// <copyright file="IWorkspace.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace
{
    public interface IWorkspace
    {
        IConfiguration Configuration { get; }

        IWorkspaceServices Services { get; }

        ISession CreateSession();
    }
}
